AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m

```diff
@@ -2650,4 +2662,69 @@ - (void)testDependencyCallUsesInjectedHttpClient {
   [httpClient stopMocking];
 }
 
+- (void)testReadAndSetUpdateTrack {
+
+  // Basic state
+  XCTAssertEqual(MSDistribute.updateTrack, MSUpdateTrackPublic);
+
+  MSDistribute.updateTrack = MSUpdateTrackPrivate;
+  XCTAssertEqual(MSDistribute.updateTrack, MSUpdateTrackPrivate);
+  MSDistribute.updateTrack = MSUpdateTrackPublic;
+  XCTAssertEqual(MSDistribute.updateTrack, MSUpdateTrackPublic);
+}
+
+- (void)testSetInvalidUpdateTrack {
+
+  // If
+  MSDistribute.updateTrack = MSUpdateTrackPrivate;
+
+  // When
+  MSDistribute.updateTrack = 100;
+
+  // Then
+  XCTAssertEqual(MSDistribute.updateTrack, MSUpdateTrackPrivate);
+}
+
+- (void)testForceSwitchToPublicUpdate {
+
+  // If
+  id utilityMock = [self mockMSPackageHash];
+  id distributeMock = OCMPartialMock(self.sut);
+  OCMReject([distributeMock requestInstallInformationWith:OCMOCK_ANY]);
+  OCMStub([distributeMock checkLatestRelease:OCMOCK_ANY distributionGroupId:OCMOCK_ANY releaseHash:OCMOCK_ANY]).andDo(nil);
+  [MSMockKeychainUtil mockStatusCode:errSecItemNotFound forKey:kMSUpdateTokenKey];
+  OCMStub([distributeMock canBeUsed]).andReturn(YES);
+
+  // When
+  [distributeMock setUpdateTrack:MSUpdateTrackPublic];
+
+  // Then
+  OCMVerify([distributeMock checkLatestRelease:OCMOCK_ANY distributionGroupId:OCMOCK_ANY releaseHash:OCMOCK_ANY]);
+
+  // Clear
+  [utilityMock stopMocking];
+  [distributeMock stopMocking];
+}
+
+- (void)testForceSwitchToPrivateUpdate {
+
+  // If
+  id utilityMock = [self mockMSPackageHash];
+  id distributeMock = OCMPartialMock(self.sut);
+  OCMStub([distributeMock checkLatestRelease:OCMOCK_ANY distributionGroupId:OCMOCK_ANY releaseHash:OCMOCK_ANY]).andDo(nil);
+  [MSMockKeychainUtil mockStatusCode:errSecItemNotFound forKey:kMSUpdateTokenKey];
+  OCMStub([distributeMock canBeUsed]).andReturn(YES);
+  OCMStub([distributeMock sharedInstance]).andReturn(distributeMock);
+
+  // When
+  [distributeMock setUpdateTrack:MSUpdateTrackPrivate];
+
+  // Then
+  OCMVerify([distributeMock requestInstallInformationWith:OCMOCK_ANY]);
+
+  // Clear
+  [utilityMock stopMocking];
+  [distributeMock stopMocking];
+}
+
```

Comment:
There are a few more unit test scenarios covered in android PR (https://github.com/microsoft/appcenter-sdk-android/pull/1353/files#diff-144a2237ac809ebb37aedb1c3c3724abR71):
- invalidPersistedTrackFallsBackToPublic
- persistValueWhenDisabled
- persistValueWhenSwitchBeforeStart
- switchTrackBeforeCallCompletes
- switchTrackAfterCallCompletes
- switchTrackBeforeForeground
- switchToSameTrackDoesNothing

No need to achieve 100% parity with android on the unit test level, but I think it's worth to add whatever makes sense for iOS.